platform/tests/TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -510,6 +511,31 @@ static control_t test_printf_x(const size_t call_count)
     result_baseline = make_test_string("x: ", 11259375, BASE_16, "\r\n");
     TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
 
+    // Test prepending zeros and width size
+    result_minimal = mbed_printf("04X: %04X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "04X: %04X\r\n", 0x04F6);
+    result_baseline = sizeof("04X: 04F6\r\n") - 1;
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_file);
+
+    result_minimal = mbed_printf("0X: %0X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "0X: %0X\r\n", 0x04F6);
+    result_baseline = sizeof("0X: 4F6\r\n") - 1;
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_file);
+
+    result_minimal = mbed_printf("6X: %6X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "6X: %6X\r\n", 0x04F6);
+    result_baseline = sizeof("6X: 4F6\r\n") - 1;
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_file);
+
+    result_minimal = mbed_printf("4X: %4X\r\n", 0xFF04F6);
+    result_file = mbed_fprintf(stderr, "4X: %4X\r\n", 0xFF04F6);
+    result_baseline = sizeof("4X: 4F6\r\n") - 1;
```

Comment:
This is not the expected output. It should be "FF04F6"

Reply:
But `%4X` means only four hex characters, and it's not `%04X` meaning 0-padding is off, so the result is `4F6` I think?

Reply:
@evedon is right. The standard printf treats `%4X`/`%.4X` as four hex characters _minimum_, if you actually try that. @EmbedEngineer

Reply:
https://en.wikipedia.org/wiki/Printf_format_string#Width_field
"The Width field specifies a minimum number of characters to output, and is typically used to pad fixed-width fields in tabulated output, where the fields would otherwise be smaller, although it does not cause truncation of oversized fields."

Reply:
@evedon @LDong-Arm you are right. I will try to fix that asap but unfortunately I am quite busy at the moment, so it might take some time.